crates/miden-objects/src/lib.rs

```diff
@@ -73,6 +76,10 @@ pub mod assembly {
         diagnostics,
         mast,
     };
+
+    pub fn default_source_manager_arc_dyn() -> Arc<dyn SourceManagerSync> {
+        Arc::new(DefaultSourceManager::default())
+    }
```

Comment:
This feels a bit out of place, but I also get that writing this again and again is annoying. I wonder if this would best be defined as `DefaultSourceManager::default_arc` - that way at least it's not a freestanding function but more clearly associated with its type. This would have to be preferably done in the VM and alternatively as an extension trait in base.

Reply:
https://github.com/0xMiden/miden-vm/issues/2088